File: MB-Secretaria/observability_py/escalation_metrics.py
from .types import ObservabilityEvent


def calculate_escalation_metrics(events: list[ObservabilityEvent]) -> dict[str, int]:
    scoped = [event for event in events if "escalation" in event.event_type]
    return {
        "escalation_count": len(scoped),
        "human_review_escalations": len([event for event in scoped if event.layer == "review"]),
        "confidence_escalations": len([event for event in scoped if event.layer == "confidence"]),
        "critical_escalations": len([event for event in scoped if event.severity == "critical"]),
    }

File: MetaBrain/observability/escalation-metrics.ts
import type { ObservabilityEvent } from "./types";

export function calculateEscalationMetrics(events: ObservabilityEvent[]) {
  const escalationEvents = events.filter((event) => event.event_type.includes("escalation"));
  return {
    escalation_count: escalationEvents.length,
    human_review_escalations: escalationEvents.filter((event) => event.layer === "review").length,
    confidence_escalations: escalationEvents.filter((event) => event.layer === "confidence").length,
    critical_escalations: escalationEvents.filter((event) => event.severity === "critical").length,
  };
}
